File: TLBB1.5/Client/Assets/Client/Scripts/Table/Table_GambleCost.cs
//This code create by CodeEngine mrd.cyou.com ,don't modify
using System;
 using System.Collections.Generic;
 using System.Collections;

namespace GCGame.Table{

[Serializable]
 public class Tab_GambleCost : ITableOperate
{ private const string TAB_FILE_DATA = "gamble_cost.txt";
 public enum _ID
 {
 INVLAID_INDEX=-1,
ID_ONE_COST,
ID_TEN_COST,
MAX_RECORD
 }
 public string GetInstanceFile(){return TAB_FILE_DATA; }

private int m_OneCost;
 public int OneCost { get{ return m_OneCost;}}

private int m_TenCost;
 public int TenCost { get{ return m_TenCost;}}

public bool LoadTable(Hashtable _tab)
 {
 if(!TableManager.ReaderPList(GetInstanceFile(),SerializableTable,_tab))
 {
 throw TableException.ErrorReader("Load File{0} Fail!!!",GetInstanceFile());
 }
 return true;
 }
 public void SerializableTable(ArrayList valuesList,string skey,Hashtable _hash)
 {
 if (string.IsNullOrEmpty(skey))
 {
 throw TableException.ErrorReader("Read File{0} as key is Empty Fail!!!", GetInstanceFile());
 }

 if ((int)_ID.MAX_RECORD!=valuesList.Count)
 {
 throw TableException.ErrorReader("Load {0} error as CodeSize:{1} not Equal DataSize:{2}", GetInstanceFile(),_ID.MAX_RECORD,valuesList.Count);
 }
 Int32 nKey = Convert.ToInt32(skey);
 Tab_GambleCost _values = new Tab_GambleCost();
 _values.m_OneCost =  Convert.ToInt32(valuesList[(int)_ID.ID_ONE_COST] as string);
_values.m_TenCost =  Convert.ToInt32(valuesList[(int)_ID.ID_TEN_COST] as string);

 _hash.Add(nKey,_values); }


}
}

File: TLBB1.5/tabletools/CodeTable/Java/xjgame/table/Table_GambleCost.java
//This code create by CodeEngine mrd.cyou.com ,don't modify
package xjgame.table;
import java.util.ArrayList;
 import java.util.HashMap;

public class Table_GambleCost implements ITableOperate {
 public final static String TAB_FILE_DATA = "/gamble_cost.txt";
 public static final int ID_ONE_COST = 0 ;
public static final int ID_TEN_COST = 1 ;
public static final int MAX_RECORD = 2 ;

 public String GetInstanceFile(){return TAB_FILE_DATA; }

private int m_OneCost;
 public int GetOneCost() { return m_OneCost;}

private int m_TenCost;
 public int GetTenCost() { return m_TenCost;}

public boolean LoadTable(HashMap<Integer, Table_GambleCost> _tab) throws TableException
 {
 if(!TableManager.ReaderPList(GetInstanceFile(),this,_tab))
 {
 throw TableException.ErrorReader("Load File{0} Fail!!!",GetInstanceFile());
 }
 return true;
 }

 @Override
 public void SerializableTable(ArrayList<String> valuesList,String skey,Object obj) throws TableException {
 @SuppressWarnings("unchecked")
 HashMap<Integer, Table_GambleCost> _hash = (HashMap<Integer, Table_GambleCost> )obj;
 if (skey.isEmpty())
 {
 throw TableException.ErrorReader("Read File"+GetInstanceFile()+" as key is Empty Fail!!!");
 }

 if (MAX_RECORD!=valuesList.size())
 {
 throw TableException.ErrorReader("Load "+GetInstanceFile()+" error as CodeSize:"+MAX_RECORD+" not Equal DataSize:"+valuesList.size());
 }
 try { 
 Integer nKey = Integer.parseInt(skey);
 Table_GambleCost _values = new Table_GambleCost();
 _values.m_OneCost =  Integer.parseInt((String)valuesList.get(ID_ONE_COST));
_values.m_TenCost =  Integer.parseInt((String)valuesList.get(ID_TEN_COST));

 _hash.put(nKey,_values);
 } catch (NumberFormatException e) {
 throw TableException.ErrorReader("Load "+GetInstanceFile()+" error as :" + e.getMessage());
 }
 }


}

